libs/arcade-cli/arcade_cli/authn.py

```diff
@@ -1,35 +1,747 @@
+"""
+OAuth authentication module for Arcade CLI.
+
+Implements OAuth 2.0 Authorization Code flow with PKCE for secure CLI authentication.
+Uses authlib for OAuth protocol handling.
+"""
+
 import os
+import secrets
 import threading
+import uuid
+import webbrowser
+from collections.abc import Generator
+from contextlib import contextmanager
+from dataclasses import dataclass
+from datetime import datetime, timedelta
 from http.server import BaseHTTPRequestHandler, HTTPServer
-from typing import Any
+from pathlib import Path
+from typing import Any, Callable
 from urllib.parse import parse_qs
 
+import httpx
 import yaml
-from arcade_core.constants import (
-    ARCADE_CONFIG_PATH,
-    CREDENTIALS_FILE_PATH,
-)
+from arcade_core.config_model import AuthConfig, Config, ContextConfig, UserConfig
+from arcade_core.constants import ARCADE_CONFIG_PATH, CREDENTIALS_FILE_PATH
+from authlib.integrations.httpx_client import OAuth2Client
+from jinja2 import Environment, FileSystemLoader
+from pydantic import AliasChoices, BaseModel, Field
 from rich.console import Console
 
-from arcade_cli.constants import (
-    LOGIN_FAILED_HTML,
-    LOGIN_SUCCESS_HTML,
-)
+from arcade_cli.constants import PROD_COORDINATOR_HOST
+
+# Set up Jinja2 templates
+_TEMPLATES_DIR = Path(__file__).parent / "templates"
+_jinja_env = Environment(loader=FileSystemLoader(_TEMPLATES_DIR), autoescape=True)
+
+
+def _render_template(template_name: str, **context: Any) -> bytes:
+    """Render a Jinja2 template and return as bytes."""
+    template = _jinja_env.get_template(template_name)
+    return template.render(**context).encode("utf-8")
```

Comment:
We previously returned an old and hardcoded HTML page on login success/failure. I brought our newer and nicer jinja templates over from Coordinator.